Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,676 @@
+        # for torrent in self.torrents.itervalues():
+        #     self.stop_download(torrent)
+
+    def get_source_object(self, sourcekey):
+        """
+        Get the actual object of the source string/key
+        """
+        # first, assume it's already source-friendly format
+        ret = self.boosting_sources.get(sourcekey, None)
+
+        # we can't find source, lets assume it's string now
+        if not ret:
+            try:  # is it channel?
+                int(sourcekey, 16)
+                ret = self.boosting_sources.get(unhexlify(sourcekey), None)
+            except ValueError:
+                # it's not, so we can't find it
+                self._logger.debug("Can't find source object in type %s", type(sourcekey))
+
+        return ret
+
+    def set_enable_mining(self, source, mining_bool=True, force_restart=False):
+        """
+        Dynamically enable/disable mining source.
+        """
+        for ihash, tor in self.torrents.iteritems():
+            if tor['source'] == source_to_string(source):
+                self.torrents[ihash]['enabled'] = mining_bool
+
+                # pause torrent download from disabled source
+                if not mining_bool:
+                    self.stop_download(tor)
+
+        self.boosting_sources[string_to_source(source)].enabled = mining_bool
+
+        self._logger.info("Set mining source %s %s", source, mining_bool)
+
+        if force_restart:
+            self._select_torrent()
+
+    def add_source(self, source):
+        """
+        add new source into the boosting manager
+        """
+        if source not in self.boosting_sources:
+            args = (self.session, source, self.settings, self.on_torrent_insert)
+
+            isdir = os.path.isdir(source)
+
+            if isdir:
+                self.boosting_sources[source] = DirectorySource(*args)
+            elif source.startswith('http://') or source.startswith('https://'):
+                self.boosting_sources[source] = RSSFeedSource(*args)
+            elif len(source) == 20:
+                self.boosting_sources[source] = ChannelSource(*args)
+            else:
+                self._logger.error("Cannot add unknown source %s", source)
+                return
+
+            self._logger.info("Added source %s", source)
+        else:
+            self._logger.info("Already have source %s", source)
+
+    def remove_source(self, source_key):
+        """
+        remove source by stop the downloading and remove its metainfo for all its swarms
+        """
+        if source_key in self.boosting_sources:
+            source = self.boosting_sources.pop(source_key)
+            source.kill_tasks()
+            self._logger.info("Removed source %s", source_key)
+
+            rm_torrents = [torrent for _, torrent in self.torrents.items()
+                           if torrent['source'] == source_to_string(source_key)]
+
+            for torrent in rm_torrents:
+                self.stop_download(torrent)
```

Comment:
Yeah, this function call `stop_download` which remove the download from session.